_chapters/parsercombinators.md

```diff
@@ -225,11 +225,14 @@ is c = do
   next c
 ```
 
-And finally our `Parser` for trying to apply a first `Parser`, and then an alternate `Parser` if the first fails.  This allows us to encode the alternatives in our BNF grammar rules.
+And finally we introduce the `Alternative` typeclass for our `Parser` for trying to apply a first `Parser`, and then an alternate `Parser` if the first fails.  This allows us to encode the alternatives in our BNF grammar rules.
 
 ```haskell
-(|||) :: Parser a -> Parser a -> Parser a
-p1 ||| p2 = P (\i -> let f (Error _) = parse p2 i
+instance Alternative Parser where
+  empty :: Parser a
+  empty = Parser $ const (Error UnexpectedEof)
```

Comment:
Not sure what error to put, parsec puts unknownError but we do not have that in our definition at the moment